x/auth/vesting/types/vesting_account_test.go

```diff
@@ -924,3 +925,585 @@ func initBaseAccount() (*authtypes.BaseAccount, sdk.Coins) {
+			expectedEndTime:  now.Add(25 * time.Hour).Unix(),
+			testTime:         now.Unix(), // Test time is before start time
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			bacc, _ := initBaseAccount()
+			cva, err := types.NewContinuousVestingAccount(bacc, tc.originalVesting, tc.startTime, tc.endTime)
+			require.NoError(t, err)
+
+			// Setup delegations
+			cva.DelegatedVesting = tc.delegatedVesting
+			cva.DelegatedFree = tc.delegatedFree
+
+			// Only test future vesting if the vesting period hasn't completed yet and has a non-zero duration
+			if tc.endTime > tc.testTime && tc.endTime > tc.startTime {
+				// Check vested coins at future time points BEFORE update
+				// Calculate several future checkpoints to verify vesting progression
+				duration := tc.endTime - tc.startTime
+				checkpoints := []float64{0.25, 0.5, 0.75, 1.0} // 25%, 50%, 75%, 100% of vesting period
+
+				// Store pre-update vested amounts for each checkpoint
+				preUpdateVestedCoins := make([]sdk.Coins, len(checkpoints))
+				for i, fraction := range checkpoints {
+					// Calculate the checkpoint time
+					checkpointTime := tc.startTime + int64(float64(duration)*fraction)
+					if checkpointTime > tc.testTime {
+						// Only track future checkpoints (after our test time)
+						preUpdateVestedCoins[i] = cva.GetVestedCoins(time.Unix(checkpointTime, 0))
+					}
+				}
+
+				// Update the vesting schedule
+				err = cva.UpdateSchedule(tc.rewardCoins)
+				require.NoError(t, err)
+
+				// Verify results
+				require.Equal(t, tc.expectedVesting, cva.OriginalVesting)
+				require.Equal(t, tc.expectedEndTime, cva.EndTime)
+
+				// Check vested coins at the same future time points AFTER update
+				for i, fraction := range checkpoints {
+					checkpointTime := tc.startTime + int64(float64(duration)*fraction)
+					if checkpointTime > tc.testTime {
+						postUpdateVestedCoins := cva.GetVestedCoins(time.Unix(checkpointTime, 0))
+
+						// For cases where rewards are added (expectedVesting > originalVesting)
+						if !tc.expectedVesting.Equal(tc.originalVesting) && fraction > 0 {
+							// Verify more coins vest at each checkpoint after the update
+							for _, coin := range postUpdateVestedCoins {
+								// Find matching coin in pre-update vested coins
+								preUpdateAmt := math.ZeroInt()
+								for _, preCoin := range preUpdateVestedCoins[i] {
+									if preCoin.Denom == coin.Denom {
+										preUpdateAmt = preCoin.Amount
+										break
+									}
+								}
+
+								// If this denomination had vesting coins before the update
+								if !preUpdateAmt.IsZero() {
+									// Verify more coins vest at this checkpoint after the update
+									require.True(t, coin.Amount.GT(preUpdateAmt),
+										"Checkpoint %v (%v%%): Expected more vested coins after update. Got %v, expected more than %v",
+										checkpointTime, fraction*100, coin, preUpdateAmt)
+								}
+							}
+						}
+					}
+				}
```

Comment:
What is happening here? This is so indented it's not easy to follow. Can this please be refactored?

Reply:
this checks that the amounts unlocked at stages after an update schedule is called is greater than the original. can look at simplifying